File: Programy z Materiałów (PDF'ów)/Python/Dzialania w systemach innych niz dziesietny/Mnozenie w systemie dwojkowym.py
def Dodaj(a,b):
    przn = 0
    c = ""
    while len(a) < len(b):
        a = '0' + a
    while len(b) < len(a):
        b = '0' + b
    for i in range(len(a) - 1, -1, -1):
        suma = przn + int(a[i]) + int(b[i])
        if suma % 2 == 1:
            c = '1' + c
        else:
            c = '0' + c
        przn = suma // 2
    if przn == 1:
        c = '1' + c
    return c
        
c = "0"
a = str(input("Podaj liczbe a: "))
b = str(input("Podaj liczbe b: "))
for i in range(len(b)):
    if b[i] == '1':
        c = Dodaj(c,a)
    a = a + "0";
print(c)

File: Programy z Materiałów (PDF'ów)/C++/Dzialania w systemach innych niz dziesietny/Mnozenie w systemie dwojkowym.cpp
#include <iostream>
#include <string>
using namespace std;

string Dodaj(string a, string b)
{
    int i, przn = 0, suma;
    string c="";
    while (a.size() < b.size()) a = '0' + a;
    while (b.size() < a.size()) b = '0' + b;
    for (i = a.size() - 1; i>=0; i--)
    {
        suma = przn + a[i] - '0' + b[i] - '0';
        if (suma % 2 == 1) c = '1' + c;
        else c = '0' + c;
        przn = suma / 2;
    }
    if (przn == 1) c = '1' + c;
    return c;
}

int main()
{
    string a, b, c = "0";
    int i,j;
    cin >> a >> b;
    for (i = b.size() - 1; i >= 0; i--)
    {
        if (b[i] == '1') c = Dodaj(c, a);
        a = a + '0';
    }
    cout << "Wynik: " << c << endl;
    return 0;
}